courses: smoother creation workflow (fixes #8952)

Mutugiii opened the pull request:
Fixes #8952

Improve the courses add/edit draft workflow. This is more intuitive for users & accomodates the unsaved changes dialog. Previously, changes were being auto saved.

<img width="3406" height="1852" alt="image" src="https://github.com/user-attachments/assets/d91b3bf9-0c39-4441-a790-df660a3051f4" />

jessewashburn reviewed (changes requested):
It would be nice if there was a way to delete a draft. 

Also, the unsaved changes guard is firing when a course is added. 

<img width="1909" height="908" alt="image" src="https://github.com/user-attachments/assets/006d3a2b-6949-4e3b-8882-b395629d14b4" />

Mutugiii commented:
Fixed the unsaved changes dialog issue on course add

Also added a delete draft workflow

<img width="3406" height="1834" alt="image" src="https://github.com/user-attachments/assets/96f0a5ef-bfcd-49aa-a761-a0e5ab84ba0e" />
